Terminal.Gui/View/Layout/Dim.cs

```diff
@@ -0,0 +1,766 @@
+        }
+
+        var newDim = new DimCombine (true, left, right);
+        (left as DimView)?.Target.SetNeedsLayout ();
+
+        return newDim;
+    }
+
+    /// <summary>Creates an Absolute <see cref="Dim"/> from the specified integer value.</summary>
+    /// <returns>The Absolute <see cref="Dim"/>.</returns>
+    /// <param name="n">The value to convert to the pos.</param>
+    public static implicit operator Dim (int n) { return new DimAbsolute (n); }
+
+    /// <summary>
+    ///     Subtracts a <see cref="Dim"/> from a <see cref="Dim"/>, yielding a new
+    ///     <see cref="Dim"/>.
+    /// </summary>
+    /// <param name="left">The <see cref="Dim"/> to subtract from (the minuend).</param>
+    /// <param name="right">The <see cref="Dim"/> to subtract (the subtrahend).</param>
+    /// <returns>The <see cref="Dim"/> that is the <c>left</c> minus <c>right</c>.</returns>
+    public static Dim operator - (Dim left, Dim right)
+    {
+        if (left is DimAbsolute && right is DimAbsolute)
+        {
+            return new DimAbsolute (left.Anchor (0) - right.Anchor (0));
+        }
+
+        var newDim = new DimCombine (false, left, right);
+        (left as DimView)?.Target.SetNeedsLayout ();
+
+        return newDim;
+    }
+
+    #endregion operators
+
+    #region overrides
+
+    /// <summary>
+    ///     Diagnostics API to determine if this Dim object references other views.
+    /// </summary>
+    /// <returns></returns>
+    internal virtual bool ReferencesOtherViews () { return false; }
+
+    /// <inheritdoc/>
+    public override bool Equals (object other) { return other is Dim abs && abs == this; }
+
+    /// <inheritdoc/>
+    public override int GetHashCode () { return Anchor (0).GetHashCode (); }
+
+    #endregion overrides
+}
+
+/// <summary>
+///     Represents a dimension that is a fixed size.
+/// </summary>
+/// <remarks>
+///     <para>
+///         This is a low-level API that is typically used internally by the layout system. Use the various static
+///         methods on the <see cref="Dim"/> class to create <see cref="Dim"/> objects instead.
+///     </para>
+/// </remarks>
+/// <param name="size"></param>
+public class DimAbsolute (int size) : Dim
+{
+    /// <inheritdoc/>
+    public override bool Equals (object other) { return other is DimAbsolute abs && abs.Size == Size; }
+
+    /// <inheritdoc/>
+    public override int GetHashCode () { return Size.GetHashCode (); }
+
+    /// <summary>
+    ///     Gets the size of the dimension.
+    /// </summary>
+    public int Size { get; } = size;
+
+    /// <inheritdoc/>
+    public override string ToString () { return $"Absolute({Size})"; }
+
+    internal override int Anchor (int size) { return Size; }
+
+    internal override int Calculate (int location, int superviewContentSize, View us, Dimension dimension)
+    {
+        // DimAbsolute.Anchor (int size) ignores width and returns n
+        return Math.Max (Anchor (0), 0);
```

Comment:
Just thought of this right now...

Because of GetHashCode needing a proper implementation and because the name of this is goofy and non-matching, how about the backing field be named appropriately for each type (since that field should be private, that is perfectly cool). Then there won't be these odd mismatches that look like an accident if you're not familiar with the code.

Reply:
I'm not sure what you're alluding to here:

- "this is goofy and non-matching"). 
- "these odd missmatches"

Can you look at latest and clarify?

Reply:
Will do in a bit.